### PR TITLE
Genesis Validation

### LINKED ISSUE #61: PoE Genesis Validation
Fail fast when validating the genesis file. 
Prevent:
-[ ] duplicate pubkeys in gentx (poe)
-[ ] duplicate pinned contracts (twasm)
-[ ] code_ids not existing for pinned contract (twasm)
-[ ] contract address not existing for privileged contracts (twasm)

To validate a genesis.json each module's [ValidateGenesis](https://github.com/confio/tgrade/blob/v0.6.0-rc2/x/poe/module.go#L64) is called with the data only under the modules name. You can work with the GenesisState types in the modules. For PoE validation is in https://github.com/confio/tgrade/blob/v0.6.0-rc2/x/poe/types/genesis.go#L75 .
Common practice is to use/ create `ValidateBasic() error` methods and fail with the first issue.

### PR DESCRIPTION
**Pull request by pinosu**
Resolves #61 

-[X] duplicate pubkeys in gentx (poe)
-[X] duplicate pinned contracts (twasm)
-[X] code_ids not existing for pinned contract (twasm)
-[X] contract address not existing for privileged contracts (twasm)

### CODE REVIEW

**Comment by pinosu** [comment]
@alpe it seems that the **_test-system suite_** is failing because the test **_TestGenesisCodePin_** is failing, due to the checks I have added. Is there something wrong with the check or it is the test itself having some issues?
Logs:
```
Test:       	TestGenesisCodePin
Messages:   	node 0: Error: pinned codeIDs map[1:{}] not found in genesis codeIDs []: invalid genesis
```

**Review by alpe — COMMENTED** [review]
Uf, the review was pending and not submitted. Sorry for this

File: x/twasm/module.go [hunk 1]
```diff
@@ -67,11 +67,11 @@ func (b AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
...
 	var data types.GenesisState
-	err := marshaler.UnmarshalJSON(message, &data)
-	if err != nil {
-		return err
+	if err := marshaler.UnmarshalJSON(message, &data); err != nil {
+		return sdkerrors.Wrap(err, "validate genesis")
 	}
-	return nil
+
+	return data.ValidateBasic()
```
Comment (alpe) [inline x/twasm/module.go:74]: 💯  important detail!

**Review by alpe — APPROVED** [review]
Good work!

**Pull request merged**